src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.activity;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.widget.EditText;
+
+import org.hawkular.client.android.R;
+import org.hawkular.client.android.backend.BackendClient;
+import org.hawkular.client.android.backend.BackendEndpoints;
+import org.hawkular.client.android.backend.model.Environment;
+import org.hawkular.client.android.backend.model.Tenant;
+import org.hawkular.client.android.util.Android;
+import org.hawkular.client.android.util.Preferences;
+import org.jboss.aerogear.android.core.Callback;
+import org.jboss.aerogear.android.pipe.callback.AbstractActivityCallback;
+
+import java.util.List;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+import butterknife.OnClick;
+import timber.log.Timber;
+
+public final class AuthorizationActivity extends AppCompatActivity implements Callback<String> {
+    @InjectView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @InjectView(R.id.edit_host)
+    EditText hostEdit;
+
+    @InjectView(R.id.edit_port)
+    EditText portEdit;
+
+    @Override
+    protected void onCreate(Bundle state) {
+        super.onCreate(state);
+        setContentView(R.layout.activity_authorization);
+
+        setUpBindings();
+
+        setUpToolbar();
+
+        setUpDefaults();
+    }
+
+    private void setUpBindings() {
+        ButterKnife.inject(this);
+    }
+
+    private void setUpToolbar() {
+        setSupportActionBar(toolbar);
+    }
+
+    private void setUpDefaults() {
+        if (Android.isDebugging()) {
+            hostEdit.append(BackendEndpoints.Community.HOST);
```

Comment:
I don’t remember reasoning behind this honestly, but probably it was related to some bug in the past I encountered. Maybe some focus stuff. I’ll change it.